crates/starknet-devnet-types/src/rpc/contract_class.rs

```diff
@@ -107,38 +109,90 @@ impl TryFrom<ContractClass> for Cairo0Json {
+                let blockifier_contract_class: blockifier::execution::contract_class::ContractClassV1 =
+                    casm_contract_class.try_into().map_err(|_| Error::ProgramError)?;
+
+                Ok(blockifier::execution::contract_class::ContractClass::V1(
+                    blockifier_contract_class,
+                ))
             }
-            _ => Err(Error::ConversionError(crate::error::ConversionError::InvalidFormat)),
         }
     }
 }
 
-impl TryFrom<ContractClass> for blockifier::execution::contract_class::ContractClass {
+impl TryFrom<ContractClass> for blockifier::execution::contract_class::ClassInfo {
     type Error = Error;
 
     fn try_from(value: ContractClass) -> Result<Self, Self::Error> {
         match value {
             ContractClass::Cairo0(deprecated_contract_class) => {
-                Ok(blockifier::execution::contract_class::ContractClass::V0(
-                    deprecated_contract_class.try_into()?,
-                ))
+                // Set abi_length to 0 as per this conversation
+                // https://spaceshard.slack.com/archives/C03HL8DH52N/p1708512271256699?thread_ts=1707845482.455099&cid=C03HL8DH52N
+                let abi_length = 0;
+                blockifier::execution::contract_class::ClassInfo::new(
+                    &blockifier::execution::contract_class::ContractClass::V0(
+                        deprecated_contract_class.try_into()?,
+                    ),
+                    0,
+                    abi_length,
+                )
+                .map_err(|err| {
+                    Error::ConversionError(ConversionError::InvalidInternalStructure(
+                        err.to_string(),
+                    ))
+                })
             }
-            ContractClass::Cairo1(_) => {
-                let casm_contract_class = CasmContractClass::try_from(value.clone())?;
-                let blockifier_contract_class: blockifier::execution::contract_class::ContractClassV1 =
-                    casm_contract_class.try_into().map_err(|_| Error::ProgramError)?;
+            ContractClass::Cairo1(sierra_contract_class) => {
+                let sierra_program_length = sierra_contract_class.sierra_program.len();
+                // Calculated as the length of the stringified abi
+                // https://spaceshard.slack.com/archives/C03HL8DH52N/p1708512271256699?thread_ts=1707845482.455099&cid=C03HL8DH52N
+                let abi_length = if let Some(abi) = sierra_contract_class.abi.as_ref() {
+                    serde_json::to_string(abi)
+                        .map(|json_str| json_str.len())
+                        .map_err(|err| Error::JsonError(JsonError::SerdeJsonError(err)))?
+                } else {
+                    0
+                };
+
+                let casm_contract_class = CasmContractClass::from_contract_class(
+                    sierra_contract_class,
+                    true,
```

Comment:
Will there always be pythonic hints?

Reply:
I dont know when they are used. If even are needed

Reply:
I meant it more like: might this code be wrong if there should be now pythonic hints? But maybe I didn't understand it right

Reply:
Ok, I misunderstood; the boolean param is for specifying whether we want the casm to contain pythonic hints. Resolving